.semgrep.yml

```diff
@@ -151,6 +151,27 @@ rules:
         - pattern: '*aws.Time($VALUE)'
     severity: WARNING
 
+  - id: data-source-with-resource-read
+    languages: [go]
+    message: Calling a resource's Read method from within a data-source is discouraged
+    paths:
+      include:
+        - aws/data_source_aws_*.go
+    patterns:
+      - pattern: $RESOURCE_READ
+      - pattern-inside: func $FUNCNAME(...) error { ... }
```

Comment:
This will need to be updated for `diag.Diagnostics` (or a separate rule created) to support context-aware functions

Reply:
ahh true! would it be too generalized of a rule if i replace `error` with something like `$RETURNTYPE` instead?